### DISCUSSION TITLE
(ebus adapter 3.1) unable to open /dev/ttyUSB0: ERR: generic device error

### DISCUSSION THREAD

**Discussion by gentleman1979** [john30/ebusd#775] [Q&A]
Hi, 
I've got my ebus adapter 3.1 (USB) thx, John!
when I use `ebusd -f -c /tmp --logareas bus --loglevel info --lograwdata=bytes -d /dev/ttyUSB0`
I get this Informations:
`2022-12-27 13:40:22.353 [bus error] unable to open /dev/ttyUSB0: ERR: generic device error
2022-12-27 13:40:22.354 [bus notice] bus started with own address 31/36
2022-12-27 13:40:22.354 [bus notice] device invalid
`
when I type `ebusctl i`
`version: ebusd 22.4.v22.4
device: /dev/ttyUSB0
signal: no signal
reconnects: 0
masters: 1
messages: 11
conditional: 0
poll: 0
update: 4
address 31: master #8, ebusd
address 36: slave #8, ebusd`
what's wrong?

**Comment by ulda** [comment]
Hi gentleman1979

das Ding spricht meines Wissens nach beim Ausliefern das Enhanced-Protokoll, d.h. du musst
`-d ens:/dev/ttyUSB0`
angeben auf der Kommandozeile.

bye
ulda

**Comment by gentleman1979** [comment]
Hi ulda,
vielen, vielen Dank!
`version: ebusd 22.4.v22.4
update check: OK
device: /dev/ttyUSB0, enhanced
signal: acquired
symbol rate: 22
max symbol rate: 65
reconnects: 0
masters: 6
messages: 12
conditional: 0
poll: 0
update: 4
address 03: master #11
address 08: slave #11
address 10: master #2
address 30: master #3
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 70: master #4
address f1: master #10
`
Ich habe eine Wolf Gastherme. Habe nun folgendes gemacht:
unter /etc/ebusd/**de**
.
└── de
    ├── broadcast.csv
    ├── memory.csv
    ├── _templates.csv
    └── wolf
..........├── 07.csv
..........├── 08.csv
..........├── 50.csv
..........└── wolf.ods
die Daten abgelegt und die Configdatei so angepasst:
`EBUSD_OPTS="-c /etc/ebusd/ -d ens:/dev/ttyUSB0 --enablehex"`
bekomme aber mit `ebusctl scan full` done.

kanns du mir hierbei auch weiter helfen??
lg

**Comment by ulda** [comment]
Hi gentleman1979

bei homeassitant funktioniert das mit /etc nicht, ich habe alle Konfigurationsdateien in ein Verheichnis unter /config/ebusd/ gelegt:
ich habe mir eine mqtt-hassio.cfg dorthin gelegt und das git  ebusd-configuration  dorhin ausgecheckt.

also insgesamt sieht das dann bei mir so aus - mit custom command line :
`
ebusd --foreground --mqtthost=core-mosquitto --mqttport=1883 --mqttuser=addons --mqttpass=hierstehtwasanderes --scanconfig --mqttjson --latency=20 --accesslevel=* --mqttint=/config/ebusd/mqtt-hassio.cfg --mqtttopic=ebusd --lograwdatafile=/config/ebusd/raw.log --device=ens:192.168.178.52:9999 --log=all:notice --log=bus:info --log=network:info -a 30  --dumpfile=/config/ebusd/dump.bin --lograwdatafile=/config/ebusd/raw.log`

damit kann man dann per
`nc -v nomeassistant.fritz.boxl 8888`
mit den [entsprechenden Befehlen] (https://github.com/john30/ebusd/wiki/3.1.-TCP-client-commandsurl) (listen, dump, raw) mal schauen was da so über den äther geht.
das geht aber gnau so auch über ebusctl

die dump und raw dateien landen entsprechend auch im /config/ebusd ordner.
da ich noch ein Problem mit den übertragenen Paketen habe, ist bei mir der --configpath noch nicht gesetzt.

bye
ulda
